### PR TITLE
emit metrics for global rate limiting

### PR DESCRIPTION
**Pull request by analogue**
### Description
Emits metrics for global rate limiting implementation.

### Testing & Reproduction steps
- Unit tests added
- Manually exceeded rate limit to `v1/catalog/nodes` via curl and then queried `v1/agent/metrics` endpoint:
```
        {
            "Name": "consul.consul.rate_limit",
            "Count": 27,
            "Rate": 2.7,
            "Sum": 27,
            "Min": 1,
            "Max": 1,
            "Mean": 1,
            "Stddev": 0,
            "Labels": {
                "limit_type": "global/read",
                "mode": "enforcing",
                "op": "Catalog.ListNodes"
            }
        },
```

### PR Checklist

* [x] updated test coverage
* [ ] external facing docs updated
* [x] not a security concern

### CODE REVIEW

**Review by boxofrad — APPROVED** [review]

**Comment by jmurret** [comment]
connect proxy listener test has a good example of testing metrics by using an in memory sink and asserting against that:  https://github.com/hashicorp/consul/blob/main/connect/proxy/listener_test.go#L128
I've done a similar thing in memberlist and it works quite well.

File: agent/consul/rate/handler_test.go [hunk 1]
```diff
@@ -224,10 +257,67 @@ func TestHandler(t *testing.T) {
...
+func testSetupMetrics(t *testing.T) *metrics.InmemSink {
+	// Record for ages (5 mins) so we can be confident that our assertions won't
+	// fail on silly long test runs due to dropped data.
+	s := metrics.NewInmemSink(10*time.Second, 300*time.Second)
+	cfg := metrics.DefaultConfig("")
+	cfg.EnableHostname = false
+	cfg.EnableRuntimeMetrics = false
+	metrics.NewGlobal(cfg, s)
+	return s
+}
```
Comment (analogue) [inline agent/consul/rate/handler_test.go:319]: I copy/pasted this and `assertCounter(...)` from `listener_test.go`. Is there a good place to move and share common test utility functions like these?

**Comment by analogue** [comment]
@boxofrad or @jmurret Can you take one last look before I close this out? Thanks!

**Review by jmurret — APPROVED** [review]
👍 Thank you!

**Pull request merged**